src/ini.cr

```diff
@@ -65,12 +65,12 @@ class INI
   #
   # INI.build({"foo" => {"a" => "1"}}, true) # => "[foo]\na = 1\n\n"
   # ```
-  def self.build(ini, space : Bool = false) : String
+  def self.build(ini : Hash(String, Hash(String, String)) | T, space : Bool = false) : String forall T
```

Comment:
This makes no sense, saying "well, it can be a Hash or basically anything else".

Reply:
Yes, else we don't have any idea of the "shape" is supposed to have, nor other types supported.
It could also be `Hash | NamedTuple`, or we enforce that an INI must only be `Hash(String, Hash(String, String))`

Reply:
How about removing the type restriction? It can be anything that's iterable. It can even be an array of tuples. But in Crystal we don't have a way to talk about those things (no interfaces, no shapes, etc.) so it's better to remove this.

Reply:
I must bet at least an `Enumerable`, doesn't it?

Reply:
No, you can have `each` on a type, not have it be `Enumerable`, and the code will work.

Reply:
Ok, however I still don't see the use case to do this.

Reply:
The benefit is not introducing a breaking change.